packages/compiler/server/serverlib.ts

```diff
@@ -853,17 +910,26 @@ export function createServer(host: ServerHost): Server {
       ) {
         kind = CompletionItemKind.Module;
       } else {
-        const type = sym.type ?? program.checker.getTypeForNode(sym.declarations[0]);
         documentation = getDoc(program, type);
         kind = getCompletionItemKind(program, type);
         deprecated = isDeprecated(program, type);
       }
-      const item: CompletionItem = {
+      let item: CompletionItem = {
         label: label ?? key,
         documentation,
         kind,
         insertText: key,
       };
+      const typeDetails = await getTypeDetails(sym, type, program);
+      if (typeDetails) {
+        item = {
```

Comment:
Here instead of recreating the item completely you can just set the detail on the item right above.